amadeus-parquet/src/internal/record/predicates.rs

```diff
@@ -18,11 +18,11 @@ impl<K, V> MapPredicate<K, V> {
 	}
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 /// Predicate for [`Group`]s
 pub struct GroupPredicate(
 	/// Map of field names to predicates for the fields in the group
-	pub(super) LinkedHashMap<String, Option<<Value as ParquetData>::Predicate>, FxBuildHasher>,
```

Comment:
Sorry I realise I overlooked when you messaged about this. This would have a perf regression. Looks like it might be a simple fix to `hashlink`, would you be able to make that PR? We can patch until your PR is merged and a new release published:

```toml
[patch.crates-io]
hashlink = { version = "0.6", git = "...", branch = "..." }
```

Reply:
Have made the fix for this and submitted a PR: https://github.com/kyren/hashlink/pull/7

Reply:
Would you like to evaluate [ritelinked](https://github.com/ritedb/ritelinked) ? This is a version derived from `hashlink` . We use `griddle` to reduce the tail delay and have solved the serialization problem.

Reply:
Yeh sure, happy to have a look :) Do you have any example code available for a common use case you might encounter that wouldn't work directly with LinkedHashMap or BTreeMap?

Reply:
@robinbernon You can directly replace `hashlink` with `ritelinked` without any additional changes, and the APIs of the two are compatible. In fact, `ritelinked` is a `hashlink` that only provides `LinkedHashMap` and `LinkedHashSet`, we just did some work to make it more usable .